### PR TITLE
DroneCAN/UAVCANv0 add NuttX SocketCAN driver

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
Adds NuttX SocketCAN driver for the DroneCAN/UAVCANv0 driver.

Only tested with uavcannode on the ucans32k146 driver, but can be extended to any NuttX SocketCAN capable MCU.

Some todo's for in the future

- SocketCAN TX Poll support (needs on modification in NuttX to implement POLLOUT for SocketCAN)
- Implement SocketCAN filters
- Expose network stack counters (err, dropped) to the SocketCAN driver

Note: I've had to modify the uavcannode init process to initialize and use the CAN driver under the same thread, before the init thread was running from a different context hence file descriptor would be wrong when handing over to workqueue.

@AlexKlimaj this might impact your boards using uavcannode, could you verify if this change has no impact on your board?

### CODE REVIEW

File: src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+} // namespace
+
+uavcan::uint32_t CanIface::socketInit(const char *can_iface_name)
+{
+
+	struct sockaddr_can addr;
+	struct ifreq ifr;
+
+	//FIXME Change this when we update to DroneCAN with CAN FD support
+	bool can_fd = 0;
```
Comment (JacobCrabill) [inline src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp:79]: I think this answers the question I was about to ask - DroneCAN doesn't support CAN FD frames currently?  I know that Cyphal does...

**Review by AlexKlimaj — APPROVED** [review]
This appears to be working on the ARK nodes.

File: src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp (hunk 1 above)
Comment (PetervdPerk-NXP) [reply src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp:79]: Master on https://github.com/dronecan/libuavcan supports CAN FD.
But the PX4 uavcan submodule hasn't been updated to support this.

Comment (JacobCrabill) [reply src/drivers/uavcan/uavcan_drivers/socketcan/driver/src/socketcan.cpp:79]: All you need is two boards of any type that support FD, and make one of them a node and the other the FMU.  Any of the STM32H7 variants support FD (though I still haven't found time to wrap up the SocketCAN support there for PX4).

File: boards/nxp/ucans32k146/cyphal.px4board [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CONFIG_DRIVERS_CYPHAL=y
```
Comment (JacobCrabill) [inline boards/nxp/ucans32k146/cyphal.px4board:1]: Do the Kconfig settings properly disable uavcan / uavcannode for this boardconfig, or do the two modules co-exist nicely?

**Review by dagar — APPROVED** [review]

File: boards/nxp/ucans32k146/cyphal.px4board (hunk 2 above)
Comment (PetervdPerk-NXP) [reply boards/nxp/ucans32k146/cyphal.px4board:1]: They should however there's was a bug in the SocketCAN stack that only allowed for a single listener.
I've fixed it now in NuttX master 
https://github.com/apache/incubator-nuttx/commit/51a845ce54e38ca6ecc13c2fd7029802cb07076f#diff-f7dcf3e1f8c366b6de10ac9320cbf4ad02b0d82cbe63dbf27d29108777d5c590
Might revisit this driver when the PX4/NuttX gets rebased and we also get the SocketCAN filter changes in.

**Pull request merged**